scripts/cdn_publish.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+version="$(git describe --abbrev=0 --tags)"
+mkdir -p cdn-dist/gravity/${version}
+
+cp dist/ui-lib/* cdn-dist/gravity/${version}
+
+versionList=""
+for version in $(git tag -l --sort=-committerdate); do
+  versionList+="<li>${version}</li>"
+done
+
+file_contents=$(<./cdn-index.template)
+echo "${file_contents//\{versionlist\}/$versionList}" > cdn-dist/gravity/index.html
+
+~/.local/bin/aws s3 sync cdn-dist s3://${CDN_BUCKET} --region=${PROD_BUCKET_REGION} --exclude "*" --include "*.html" --content-type "text/html; charset=utf-8" --cache-control "max-age=0"
+~/.local/bin/aws s3 sync cdn-dist s3://${CDN_BUCKET} --region=${PROD_BUCKET_REGION} --exclude "*.html" --cache-control "max-age=3153600000, immutable"
```

Comment:
Hmm... this kinda works - the CSS & JS files are being served up with the correct mime-types, even though we don't explicitly set them here. However, they don't have any character encoding set. Is it possible to set that to UTF-8 - at least for text-based types like CSS, JS, JSON and SVG?

While it shouldn't make much practical difference right now, we might get some weirdness if we were to add non-ASCII chars to any of them (e.g. emoji).

Reply:
Yes, it will assume file type based on extension.  If you want encoding, we need to separate them out.